v0/destinations/user/utils.js

```diff
@@ -0,0 +1,643 @@
+    const lookupFieldValue = getFieldValueFromMessage(message, lookupField);
+
+    if (lookupField === "email") {
+      return getUserByEmail(apiKey, lookupFieldValue, appSubdomain);
+    }
+
+    if (lookupField === "phone") {
+      return getUserByPhoneNumber(apiKey, lookupFieldValue, appSubdomain);
+    }
+    throw new CustomError(
+      `lookup field : ${lookupField} is not supported for this destination`,
+      400
+    );
+  } else {
+    const userId = getValueFromMessage(message, "userId");
+    if (userId) {
+      return getUserByCustomId(message, destination);
+    }
+    const email = getFieldValueFromMessage(message, "email");
+    if (isDefinedAndNotNullAndNotEmpty(email)) {
+      return getUserByEmail(apiKey, email, appSubdomain);
+    }
+    throw new CustomError("default lookup field : email value is empty", 400);
+  }
+};
+
+/**
+ * Returns 'Create User' payload
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+const createUserPayloadBuilder = (message, destination) => {
+  let commonUserPropertiesPayload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.CREATE_USER.name]
+  );
+  const { userAttributesMapping } = destination.Config;
+  const customUserAttributes = getCustomUserAttributes(
+    userAttributesMapping,
+    message
+  );
+  commonUserPropertiesPayload = {
+    ...commonUserPropertiesPayload,
+    ...customUserAttributes
+  };
+  const { endpoint } = CONFIG_CATEGORIES.CREATE_USER;
+  const method = "POST";
+  return { commonUserPropertiesPayload, endpoint, method };
+};
+
+/**
+ * Returns 'Update User' payload
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+const updateUserPayloadBuilder = (message, destination) => {
+  let commonUserPropertiesPayload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.UPDATE_USER.name]
+  );
+  const { userAttributesMapping } = destination.Config;
+  const customUserAttributes = getCustomUserAttributes(
+    userAttributesMapping,
+    message
+  );
+  commonUserPropertiesPayload = {
+    ...commonUserPropertiesPayload,
+    ...customUserAttributes
+  };
+  const { endpoint } = CONFIG_CATEGORIES.UPDATE_USER;
+  const method = "PUT";
+  return { commonUserPropertiesPayload, endpoint, method };
+};
+
+/**
+ * Returns 'Create Event Occurrence' payload
+ * @param {*} message
+ * @param {*} user
+ * @param {*} destination
+ * @returns
+ */
+const createEventOccurrencePayloadBuilder = (message, user, destination) => {
+  const { properties } = message;
+  const payload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.CREATE_EVENT_OCCURRENCE.name]
+  );
+  const { Config } = destination;
+  const { userEvents } = Config;
+  const { name } = payload;
+  const userEvent = getUserEvent(userEvents, name);
+  if (userEvent.length === 1) {
+    const [first] = userEvent;
+    payload.name = first.userEventName;
+    payload.data = getEventAttributes(first, properties);
+  }
```

Comment:
What do we do, if the user is not sending events which are mapped in the UI? Does user.com create any arbritary event if sent? If that is the case we need to handle that as well.